src/ng-xform/select-field/select-field.component.ts

```diff
@@ -18,15 +27,90 @@ import { BaseDynamicFieldComponent } from '../field-components/base-dynamic-fiel
 @Component({
   selector: 'ng-xform-select-field',
   templateUrl: './select-field.component.html',
+  providers: [{
+    provide: NG_VALUE_ACCESSOR,
+    useExisting: SelectFieldComponent,
+    multi: true
+  }],
 })
-export class SelectFieldComponent extends BaseDynamicFieldComponent<SelectField> implements OnInit {
+export class SelectFieldComponent extends BaseDynamicFieldComponent<SelectField> implements ControlValueAccessor, OnInit, AfterViewInit {
 
-  @Input() errorMessage: string;
+  @ViewChild(NgSelectComponent) select: NgSelectComponent;
+  viewModel = new BehaviorSubject<any>(null);
 
   optionValues: any[] = [];
+  optionLabel = '-';
+  typeahead: EventEmitter<string>;
+
+  _onChange: any = (value: any) => { };
+  _onTouched: any = () => { };
+
+  writeValue(obj: any): void {
+    this.viewModel.next(obj);
+  }
+
+  registerOnChange(fn: any): void {
+    this._onChange = fn;
+  }
+  registerOnTouched(fn: any): void {
+    this._onTouched = fn;
+  }
+  setDisabledState?(isDisabled: boolean): void {
+    this.select.setDisabledState(isDisabled);
+  }
+
+  updateOptionLabel() {
+    setTimeout(() => {
+      this.optionLabel = this.select.selectedItems.map(item => item.label).join(this.field.separator);
+    }, 0);
+  }
+
+  ngAfterViewInit() {
+    this.select.registerOnChange((value: any) => {
+      this.updateOptionLabel();
+      this._onChange(value);
+    });
+    this.select.registerOnTouched(this._onTouched);
+    this.viewModel.subscribe((value: any) => {
+      if (this.field.searchByValueAttributeHandler) {
```

Comment:
Why are you using `viewModel`? Can't this run directly on `writeValue` method? The answer should be documented in the class

Reply:
No, the form makes writeValue's call before the select attribute is initialized.

Reply:
Ok, so add a comment above the line `viewModel.next()` call